lib/internal/modules/esm/loader.js

```diff
@@ -159,18 +159,16 @@ function nextHookFactory(chain, meta, { validateArgs, validateOutput }) {
       // Set when next<HookName> is actually called, not just generated.
       if (generatedHookIndex === 0) { meta.chainFinished = true; }
 
-      // `context` is an optional argument that only needs to be passed when changed
-      switch (args.length) {
-        case 1: // It was omitted, so supply the cached value
-          ArrayPrototypePush(args, meta.context);
-          break;
-        case 2: // Overrides were supplied, so update cached value
-          ObjectAssign(meta.context, args[1]);
-          break;
+      const argsToApply = [args[0], args[1]];
+
+      if (argsToApply[1] == null) {
+        argsToApply[1] = meta.context;
+      } else {
+        ObjectAssign(meta.context, args[1]);
       }
 
-      ArrayPrototypePush(args, nextNextHook);
-      const output = await ReflectApply(hook, undefined, args);
+      argsToApply[2] = nextNextHook;
+      const output = await ReflectApply(hook, undefined, argsToApply);
```

Comment:
Is there a reason to be using an array instead of named args or variables, such as `arg0`, `context`, `nextNextHook`?  And then `ReflectCall` instead of `ReflectApply`?

Reply:
The names of the args are not consistent across hooks. Apply vs Call 🤷‍♂️ does it matter?

Reply:
That's the bit I wasn't sure about; `argsToApply[1] = meta.context` implies that the second argument is consistently named `context`.  The first argument does not have a consistent name, agreed, though I'm not sure what breakages or confusion you imagine will happen if we name it `arg1`.  I assume avoiding the arrays is a bit faster?  Though then we're getting into optimization territory, and I would need to back that up with a benchmark.

Reply:
I think the perf difference of that would be trivial, especially because it's short-lived in memory.

Currently:

arg0 = `specifier` for resolve, `url` for load
arg1 = same (both `context`
arg2 = next<HookName>

I think giving them specific var names is more trouble than worth, but not a strong opinion.

Reply:
For the sake of review, the most recent revision on this gist demos the code change to switch from array -> named args.  Github's PR "suggestion" feature won't accept suggestions that fall outside of the PR's changed.
https://gist.github.com/cspotcode/355b670fee02bf3b636c8f9d56d921ed/revisions

Reply:
Are we looking at the same diff?  It's only 8 lines touched, and reduces total lines of code compared to what you have.

Reply:
FWIW `ReflectCall` is not a thing.

Reply:
Oh right, so it would simplify to `hook(arg0, context, nextNextHook)`?

Reply:
oh wow. those are revisions 🤦‍♂️

Reply:
I blame Github; I couldn't find any way to link to a single commit in a gist; I'm pretty sure they removed that feature and I don't know why.

Reply:
Actually, what I ended up doing is almost identical, no?

Reply:
The reduction in SLOC comes from avoiding `argsToApply`.  I think it also addresses @aduh95's comments.

Reply:
I agree that would be more readable, and would skip the array allocation which is always nice.
```suggestion
      const output = await hook(arg0, context, nextNextHook);
```